plugins/modules/network_settings_intent.py

```diff
@@ -1635,9 +1684,7 @@ def update_global_pool(self, config):
             return
 
         # Pool exists, check update is required
-        obj_params = [
-            ("settings", "settings"),
-        ]
+        obj_params = self.get_obj_params("GlobalPool")
```

Comment:
global_obj_params = self.get_obj_params("GlobalPool")

If it is already set, then we can directly use global_obj_params in verify instead of getting again..